apps/src/templates/census2017/CensusForm.jsx

```diff
@@ -0,0 +1,313 @@
+import React, {Component} from 'react';
+import Button from '../Button';
+import color from "../../util/color";
+import i18n from "@cdo/locale";
+import $ from 'jquery';
+import {CSOptions, roleOptions, courseTopics, frequencyOptions, pledge} from './censusQuestions';
+
+const styles = {
+  question: {
+    fontSize: 16,
+    fontFamily: '"Gotham 5r", sans-serif',
+    color: color.charcoal,
+    paddingTop: 10,
+    paddingBottom: 5
+  },
+  personalQuestion: {
+    width:'33%',
+    float: 'left'
+  },
+  personalQuestionsBox: {
+    marginTop: 50
+  },
+  checkbox: {
+    width: 25,
+    height: 25,
+    padding: 0,
+    margin:0,
+    verticalAlign: 'bottom',
+    position: 'relative',
+    top: -1,
+    overflow: 'hidden',
+  },
+  pledgeBox: {
+    marginTop: 50,
+    marginBottom: 50
+  },
+  pledge: {
+    fontSize: 18,
+    fontFamily: '"Gotham 7r", sans-serif',
+    color: color.charcoal,
+    paddingBottom: 10,
+    paddingTop: 10
+  },
+  option: {
+    fontFamily: '"Gotham 4r", sans-serif',
+    color: color.charcoal,
+  },
+  options: {
+    marginLeft: 35
+  },
+  input: {
+    height: 30,
+    width: 250,
+    fontFamily: '"Gotham 3r", sans-serif',
+    padding: 5
+  },
+  textArea: {
+    height: 100,
+    width: '100%',
+    fontFamily: '"Gotham 3r", sans-serif',
+    padding: 5
+  },
+  thankYouBox: {
+    display: 'none',
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderColor: color.green,
+    padding: 10
+  }
+};
+
+class CensusForm extends Component {
+
+  state = {
+    selected: [],
+    submission: {
+      name: '',
+      email: '',
+      role: '',
+      followUpFrequency: '',
+      followUpMore: ''
+    }
+  };
+
+  handleChange(propertyName, event) {
+    this.setState({
+      submission: {
+        ...this.state.submission,
+        [propertyName]: event.target.value
+      }
+    });
+  }
+
+  toggle(option) {
+    const index = this.state.selected.indexOf(option);
+    if (index >= 0) {
+      this.setState(previousState => {
+        selected: previousState.selected.splice(index, 1);
+      });
+    } else {
+      this.setState(previousState => {
+        selected: previousState.selected.push(option);
```

Comment:
`splice` and `push` are bad here because they modify React state, which can lead to React failing to re-render in some circumstances. For `splice`, you can just make a copy of the array before splicing it. For `push`, you could make a copy, or try something like `[...previousState.selected, option]`.

Reply:
You can also use slice/concat instead, which are immutable operations.

Reply:
Though using the spread operator like Dave has above is prob simpler than concat.

Reply:
Is there an advantage to doing this `previousState =>` business vs. just doing
```
this.setState({
  selected: [...this.state.selected, option]
})
```
My suspicion is that the pattern you're using may be necessary for cases where we can call `setState` multiple times before we rerender, but I don't know that this is the situation here.